db/perf_context_test.cc

```diff
@@ -813,6 +813,77 @@ TEST_F(PerfContextTest, PerfContextByLevelGetSet) {
   ASSERT_NE(std::string::npos,
             zero_excluded.find("block_cache_miss_count = 4@level1, 2@level3"));
 }
+
+TEST_F(PerfContextTest, CPUTimer) {
+  DestroyDB(kDbName, Options());
+  auto db = OpenDb();
+  WriteOptions write_options;
+  ReadOptions read_options;
+  SetPerfLevel(PerfLevel::kEnableTimeAndCPUTimeExceptForMutex);
+
+  for (int i = 0; i < FLAGS_total_keys; ++i) {
+    std::string key = "k" + ToString(i);
+    std::string value = "v" + ToString(i);
+
+    db->Put(write_options, key, value);
+  }
+
+  {
+    // Get
+    get_perf_context()->Reset();
+    std::string value;
+    ASSERT_OK(db->Get(read_options, "k0", &value));
+    ASSERT_EQ(value, "v0");
+
+    if (FLAGS_verbose) {
+      std::cout << "Get CPU time nanos: " << get_perf_context()->get_cpu_nanos << "ns\n";
```

Comment:
it would be good to add some verification logic in the test, printing isn't very useful at catching bugs

Reply:
> does the counter increases monotonously

Yeah I can check that by doing multiple operations at the end of the test.

> how does the counter behave when multiple iterators coexist

It seems to me perf_context is thread safe if thread local is supported, but I could be wrong.

> how does the counter behave when iterators are created/destroyed

Can you elaborate? Did you mean we should expect no-op to counters at iterator creation/destruction?

Reply:
what I meant was iterator creation/deletion should not affect the counter, and the counter should increase normally when there are multiple iterators doing seek/next, just some test ideas to consider, you can make your own judgement